### PR TITLE
add documentation

### PR DESCRIPTION
**Pull request by daschw**

### CODE REVIEW

**Comment by daschw** [comment]
> Makes sense to me. I'm not very active here anymore, except in docs (the docs are still hosted on my servers)

_Originally posted by @ChrisRackauckas in https://github.com/JuliaPlots/Plots.jl/issues/2478#issuecomment-600172473_

@ChrisRackauckas do you know if automatically forwarding from `juliaplots.github.io/` to `http://docs.juliaplots.org/latest/` might be an issue here?
At least the preview from this PR does not seem to work: https://juliaplots.github.io/RecipesBase.jl/previews/PR64/index.html

**Comment by asinghvi17** [comment]
That may be because you're on an external fork.  PR previews only work on branches within the main repo, because they require using the private key - which could conceivably be leaked by a malicious PR.

**Comment by asinghvi17** [comment]
I can confirm that the previews work on MakieGallery, for what it's worth - that is hosted at `makie.juliaplots.org`.

**Comment by daschw** [comment]
OK, thanks for the info! So I will only know, if this works, after I merge?

File: docs/make.jl [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        "Home" => "index.md",
+        "Recipe Syntax" => "syntax.md",
+        "Recipe Types" => "types.md",
+        "Internals" => "internals.md",
+        "Library" => "api.md"
+    ],
+    modules = [RecipesBase]
+)
+
+# deploydocs(
```
Comment (asinghvi17) [inline docs/make.jl]: You will have to uncomment this ;)

Comment (daschw) [reply docs/make.jl]: Ha, thanks! :man_facepalming:

**Comment by daschw** [comment]
I get a 404 for the docs link. Any ideas how to fix this are very much appreciated.

**Comment by asinghvi17** [comment]
https://travis-ci.org/github/JuliaPlots/RecipesBase.jl/jobs/668377432?utm_medium=notification&utm_source=github_status

Looks like the doc build just started.

**Comment by asinghvi17** [comment]
https://travis-ci.org/github/JuliaPlots/RecipesBase.jl/jobs/668377432?utm_medium=notification&utm_source=github_status

Looks like the doc build just started.

**Comment by daschw** [comment]
Yeah, I messed something up with the DOCUMENTER_KEY and restarted

**Comment by daschw** [comment]
OK, it worked now.

**Pull request merged**